ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -20,13 +20,12 @@
 #include "Platforms/General/PFjoystick.h"
 #include <string>
 
-namespace {
+using enigma::joysticks;
 
-SDL_Joystick *joystick1 = NULL;
-SDL_Joystick *joystick2 = NULL;
+namespace {
 
-static inline SDL_Joystick* joystick_get_handle(int id) {
-  return (id - 1 < 0.5) ? joystick1 : joystick2;
+static inline SDL_Joystick *joystick_get_handle(int id) {
+  return (id < joysticks.size()) ? ((id < 1) ? joysticks[0] : joysticks[id - 1]) : joysticks[joysticks.size() - 1];
```

Comment:
I'm not a big fan of this check I would prefer you instead do
```cpp
#ifdef DEBUG_MODE
if (id < 0 || id >= joysticks.size())) { DEBUG_MESSAGE("invalid js id"); return nullptr}
#endif
return joysticks[id];
```

You could also use an AssetArray which does these kind of checks for you. Either would be fine with me.

Reply:
To remain GameMaker compatible we will need the joystick id to start at 1, not zero, other than that, I 100% agree with you. I don't like it starting at 1 though, I wish GM didn't do that.

Reply:
You can change my code to reflect gm's off by 1 bug like so

```cpp
#ifdef DEBUG_MODE
if (id < 1 || id > joysticks.size())) { DEBUG_MESSAGE("invalid js id"); return nullptr}
#endif
return joysticks[id-1];```
but please comment why you do that in the src.

However, I'm not a big fan of emulating GM's bugs in our code base just to maintain compatibility. I think a good compromise might be adding a function or variable to set first js id offset id based on some sort gm compatibility settings / extensions we add.

ie something like

```cpp 
namespace enigma {
const size_t js_offset = 1;
}
````
```cpp
#include whater_js.h

inline SDL_Joystick *joystick_get_handle(int id) {
  #ifdef DEBUG_MODE
  if (id < js_offset || id >= joysticks.size() + js_offset)) { DEBUG_MESSAGE("invalid js id"); return nullptr}
  #endif
  return joysticks[id-js_offset];```
}
```
However, I don't really expect you to handle this in this pr so just leaving a comment for now explaining why you did what you did will suffice for now I guess.